Snowflake.Data/Core/Tools/FileOperations.cs

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+using System.IO;
+
+namespace Snowflake.Data.Core.Tools
+{
+    internal class FileOperations
```

Comment:
Why this all classes are singletons? Maybe it would be good enough if they provide static functions? They don't have any state.

Reply:
The classes are singletons to be able to mock them in tests. Suppose you would like to test scenario when a config file exists in home path or temporary path. If I really would create a file I would interfere with all the tests which are touching easy logic feature. I don't see any props, but a lot of cons.

Reply:
So maybe this shouldn't be statics at all? Maybe this should be some interface and implementation passing all around? Mocking global, static methods especially when we have tests parallelisation in mind is very tricky...

Reply:
it is not static, we discussed that it's ok